geopandas/tools/sjoin.py

```diff
@@ -151,8 +177,22 @@ def _basic_checks(left_df, right_df, how, lsuffix, rsuffix):
     if not _check_crs(left_df, right_df):
         _crs_mismatch_warn(left_df, right_df, stacklevel=4)
 
+    if on_attribute:
+        for attr in on_attribute:
+            if not ((attr in left_df) and (attr in right_df)):
+                raise ValueError(
+                    "Expected column {} is missing from one of the dataframes".format(
```

Comment:
Use f-string please.

Reply:
Thanks, I still do a lot of python2 at work I forget there are much nicer ways of doing things now.